packages/vue-components/src/SearchbarPageItem.vue

```diff
@@ -1,8 +1,10 @@
 <template>
   <div v-if="item.heading" class="heading">
+    <i class="fa fa-hashtag"></i>
     <div class="heading-text">
-      <span v-html="item.heading.text"></span>
+      <span v-html="highlight(item.heading.text, value)"></span>
     </div>
+    <!--
```

Comment:
> You may also need to check if this plays correctly and nicely with [keywords](https://markbind.org/userGuide/makingTheSiteSearchable.html#keywords).
> > I've also tested it with keywords (page and headings), and based on my tests it works fine.

I actually meant moving the keywords to a separate location (e.g. below the heading as smaller, muted text), not removing them entirely.

But this looks ok to me as well (as long as it turns up).

Let's remove the dead code (one more below as well)

On a related note https://github.com/MarkBind/markbind/issues/205 would change the design completely again... but don't think I'm going to be able to complete that anytime soon.

Reply:
Ah sorry for misunderstanding! I've added in the keywords to the search results, looks like this:

![keywords long](https://user-images.githubusercontent.com/48304907/183283270-c49716c5-eb02-4c82-861c-20c04337dc5c.png)

Have also removed the dead code (:

Reply:
Hi @weiquu, thanks for the update. A gentle reminder to provide testing instructions in the future, so that reviewers can get back to you quicker:) For the screenshot you posted above, I tried it on the PR preview in https://deploy-preview-1968--markbind-master.netlify.app/ and then realized that it won't work because you have added some keywords locally to show this behavior.

Here's some screenshots of existing keywords and how they look like with this update:
- hide minimized panels
![image](https://user-images.githubusercontent.com/41845017/183283705-40038efe-f1ff-4319-b472-5666a2b9b645.png)
- regress
![image](https://user-images.githubusercontent.com/41845017/183283717-e8d459fa-bef9-4635-8645-d2401e1c8774.png)

Are the words suppose to start on a newline?

Lastly, may I double check if the content here is still valid after your change @weiquu ?
https://markbind.org/userGuide/makingTheSiteSearchable.html
Specifically:
- does keyword still work for heading and an entire page?
- does the `.always-index or .no-index` still work,  in particular, if the heading is the very first one?

Reply:
Thanks for the reminder! Will update the testing instructions.

For the newline issue, can I check if you went into the PR preview and directly typed in the search words? I tried and there's no newline for me (e.g. regress below).
![regress](https://user-images.githubusercontent.com/48304907/183307482-c960055f-a9e8-46d4-a616-330dc7a64321.png)

I've checked through the content in the page linked, responding to the 2 pointed out:

For the `.always-index` and `.no-index`, I've tested with the code below:
```
<span id="title" class="d-none">Using Plugins</span>

<frontmatter>
  title: "Using Plugins"
</frontmatter>

###### Using Plugins {.always-index}

# Using Plugins {.no-index}
```
With this code, both don't show up (as expected), but if the positions of the 2 headings are swapped around (h1 then h6), then both don't show up again since the h6 heading is taken as the first indexed heading.

For the keywords, they do also work with the heading. However, I'm not too sure if I'm doing something wrong with the  page heading since it doesn't seem to work even if I revert back to the main build. With this code, I'm not seeing anything coming up with search word `plugin` even though I believe the expected result should be seeing the page title `Formatting Contents`:
```
<span id="title" class="d-none">Formatting Contents</span>

<frontmatter>
  title: "Formatting Contents"
  keywords: plugin
</frontmatter>

# Formatting Contents (Advanced)
```

Would appreciate any advice on the above 2 issues!

Reply:
> does the .always-index or .no-index still work, in particular, if the heading is the very first one?
>
> > With this code, both don't show up (as expected), but if the positions of the 2 headings are swapped around (h1 then h6), then both don't show up again since the h6 heading is taken as the first indexed heading.

I might be lacking some context here but imo this change should have nothing to do with these attributes. (current behaviour is fine)

E.g.

```
<frontmatter>
  title: "Using Plugins"
</frontmatter>

// Should not show. What we index =/= user friendliness of the SERP we generate.
# Using Plugins {.always-index}
```

> does keyword still work for heading and an entire page?
>
> > For the keywords, they do also work with the heading. However, I'm not too sure if I'm doing something wrong with the page heading since it doesn't seem to work even if I revert back to the main build. With this code, I'm not seeing anything coming up with search word plugin even though I believe the expected result should be seeing the page title Formatting Contents:

That's a good catch. I think we can add to the check something along the lines of `!(idx === 0 && ... && !keywords.length)`.

Reply:
On a separate note, a 🔑 icon conveys to me *security* - e.g. a password, authentication, secret key, etc. Might be confusing to readers what exactly the words following are.

I think we ought to swap up or remove the icon. 🤔 

If there's another suitable icon, let's limit it to displaying just once too so it dosen't get too repititive.

Reply:
> For the newline issue, can I check if you went into the PR preview and directly typed in the search words? I tried and there's no newline for me (e.g. regress below).

@weiquu see gif below, seems like it depends on window size:
![OQyxxAH6uv](https://user-images.githubusercontent.com/41845017/183543852-a00134c4-349d-4668-a4b2-85b02d6615b6.gif)

> On a separate note, a 🔑 icon conveys to me security

How about a 🎯 ?

> I might be lacking some context here but imo this change should have nothing to do with these attributes.

Was checking if these attributes will be affected by this implementation, because of how we  ignore duplicates (what if the duplicates have been marked with the always-index, no-index, does it still work correctly)

> For the keywords, they do also work with the heading. However, I'm not too sure if I'm doing something wrong with the page heading since it doesn't seem to work even if I revert back to the main build. 

Yes seems like it is an existing issue.

Reply:
> 🎯

This conveys target - which means a goal, okr to me personally.

Would a muted search icon work?

On further thought I'm favouring removing the icon more strongly as well, rationale is that the author can also add unicode emojis themselves iirc.

Reply:
> Was checking if these attributes will be affected by this implementation, because of how we ignore duplicates (what if the duplicates have been marked with the always-index, no-index, does it still work correctly)

🙆‍♂️